tests/test_commands.py

```diff
@@ -12,62 +10,186 @@ def datasets():
     return DatasetFactory.create_batch(3)
 
 
+@pytest.fixture
+def mock_invalid_response():
+    return [
+        {"foo": "bar"}
+    ]
+
+
 @pytest.fixture
 def mock_response(datasets):
     ds1, ds2, ds3 = datasets
     return [
         {
-            # invalid ID, should not crash the command
+            # Invalid ID, but valid reco: should not crash the command
             "id": "1",
             "recommendations": [
                 {
-                    "id": str(ds1.id)
+                    "id": str(ds1.id),
+                    "score": 50
                 }
             ]
         },
         {
-            # valid ID and recos, expect for third element
+            # valid ID and recos,
             # should process two elements w/o crashing
+            # should reorder by score and handle reco by ID and slug
             "id": str(ds2.id),
             "recommendations": [
                 {
-                    "id": str(ds1.id)
+                    "id": str(ds3.id),
+                    "score": 1
                 },
                 {
-                    "id": str(ds3.id)
+                    "id": str(ds1.slug),
+                    "score": 2
                 },
                 {
-                    "id": "%s1234" % str(ds3.id)
+                    "id": "nope",
+                    "score": 50
                 }
             ]
         },
         {
-            # empty recos, should not crash the command
+            # Valid ID but recommended dataset does not exist
             "id": str(ds3.id),
-            "recommendations": []
-        },
-        {
-            # invalid schema, should not crash the command
-            "dataset": "xxx"
+            "recommendations": [
+                {
+                    "id": "nope",
+                    "score": 50
+                },
+            ]
         }
     ]
 
 
-@pytest.mark.options(RECOMMENDATIONS_DATASETS_SOURCE_URL=MOCK_URL)
-def test_dataset_reco_command(cli, rmock, mock_response, datasets):
+@pytest.mark.usefixtures('clean_db')
+def test_clean_source(cli, datasets):
+    _ = datasets
```

Comment:
Create dummy datasets and make sure they are excluded from recommendations

Reply:
datasets are not created if you don't use the feature explicitly?

Reply:
I would probably copy `DatasetFactory.create_batch(3)` in this case, more explicit. And add a comment before.